src/rendezvous/channels/MSC3903ECDHv1RendezvousChannel.ts

```diff
@@ -0,0 +1,255 @@
+/*
+Copyright 2022 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { SAS } from '@matrix-org/olm';
+
+import {
+    RendezvousError,
+    RendezvousCode,
+    RendezvousIntent,
+    RendezvousChannel,
+    RendezvousTransportDetails,
+    RendezvousTransport,
+    RendezvousFailureReason,
+    RendezvousChannelAlgorithm,
+} from '..';
+import { encodeBase64, decodeBase64 } from '../../crypto/olmlib';
+import { crypto, subtleCrypto, TextEncoder } from '../../crypto/crypto';
+import { generateDecimalSas } from '../../crypto/verification/SASDecimal';
+
+export interface ECDHv1RendezvousCode extends RendezvousCode {
+    rendezvous: {
+        transport: RendezvousTransportDetails;
+        algorithm: RendezvousChannelAlgorithm.ECDH_V1;
+        key: string;
+    };
+}
+
+interface ECDHPayload {
+    algorithm?: RendezvousChannelAlgorithm.ECDH_V1;
+    key?: string;
+    iv?: string;
+    ciphertext?: string;
```

Comment:
Seems strange for all these to be optional, you won't get far with them missing, what's the rationale?

Reply:
Yes, please use Partial on the receive side in that case, we want to leverage types to ensure we don't miss sending the *actually required* fields

Reply:
Done.